app/transport/http/resp.go

```diff
@@ -0,0 +1,25 @@
+package http
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+)
+
+func GenerateJSONResponse(w http.ResponseWriter, statusCode int, data interface{}) {
```

Comment:
@vpakhuchyi  Please correct me if I am wrong:

We do actually need a pair of funcs (happy and sad) to build our handlers along the way: 
 ```golang
func respond(rw http.ResponseWriter,  data any, code int) {}
func respondErr(rw http.ResponseWriter, err error, code int) {}
```
and error responding worth abstraction
we may send kind of JSON as a result to frontend:
```JSON
{
  "error": {
    "code": 400,
    "reason": "user is looser"
   }
}
```

Reply:
I think we can even use the same function for that. Take a look at signatures:
``` go
func respond(rw http.ResponseWriter,  data any, code int) {}
func respondErr(rw http.ResponseWriter, err error, code int) {}
```
They are almost equal. 
I think we can use this one and pass data based on the situation - success response or error response.
In such a case this function will know nothing about data. Only knows how to write this data to the HTTP response.
``` go
func respond(rw http.ResponseWriter,  data any, code int) {}
```

Reply:
Regarding the error message format.
Do we really want to duplicate a status code? 

I would start from something like this:
```json
{
  "error": "user is looser"
}
```

In the future we can extend this format to include error codes (it's not the same as status code, messages translations etc).

Reply:
> I think we can even use the same function for that. Take a look at signatures:
> 
> ```go
> func respond(rw http.ResponseWriter,  data any, code int) {}
> func respondErr(rw http.ResponseWriter, err error, code int) {}
> ```
> 
> They are almost equal. I think we can use this one and pass data based on the situation - success response or error response. In such a case this function will know nothing about data. Only knows how to write this data to the HTTP response.
> 
> ```go
> func respond(rw http.ResponseWriter,  data any, code int) {}
> ```

If we can simplify it down to one func it is cool. I do not get how exactly we will find out that underlying data is an error and render an appropriate message?

Reply:
We have to marshal data and it's not important for us to know how does it look.
So, caller of this func is responsible for passing data in a right format. 
But this func is a bit stupid and only marshal what he received.